### PR TITLE
Added basic photo capability.

### PR DESCRIPTION
**Pull request by ProCynic**
New routes:
    GET     /photos      #view all photos
    POST    /photos      #add a photo with logged in user as owner
    GET     /photos/id   #view a specific photo
    DELETE  /photos/id   #delete a photo
    GET     /users/id/photos #get all photos for a specific user

New model Photo.java has image blob and User owner.

New view template photos.html.  Basic skeleton, needs work.

New Application.java methods:
    photos(Long id) //view all photos, or all photos by a specific user
    getPhoto(Long id)   //render a specific photo
    addPhoto(Photo photo) // add a photo with the logged in user as owner
    removePhoto(Long id) // delete a photo

### CODE REVIEW

File: conf/routes [hunk 1]
```diff
@@ -11,6 +11,13 @@ GET     /                                       Application.news
 GET     /about                                  Application.about
 GET     /account                                Application.account
 
+# Photos
+GET     /photos                                 Application.photos(id:null)
+POST    /photos                                 Application.addPhoto
+GET     /photos/{<[0-9]+>id}                    Application.getPhoto
+DELETE  /photos/{<[0-9]+>id}                    Application.removePhoto
```
Comment (ProCynic) [inline conf/routes]: Needs authentication.

File: app/models/Photo.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package models;
```
Comment (amshali) [inline app/models/Photo.java:1]: Please add dateAdded and caption to this entity.

File: app/controllers/Application.java [hunk 3]
```diff
@@ -177,9 +177,39 @@ public static void deleteComment(Long id, Long userId) {
     c.delete();
     news(userId);
   }
-
```
Comment (amshali) [inline app/controllers/Application.java:180]: Let's refactor the functionalities  for photo  into a different class in controller.

File: app/controllers/Application.java [hunk 4]
```diff
@@ -177,9 +177,39 @@ public static void deleteComment(Long id, Long userId) {
...
+  
   public static void postComment(Long postId, String author, String content) {
     Post post = Post.findById(postId);
     post.addComment(author, content);
   }
-}
+
+  public static void photos(Long id) {
```
Comment (amshali) [inline app/controllers/Application.java]: Please change the name of id parameter to ownerId or userId.

File: app/controllers/Application.java [hunk 5]
```diff
@@ -177,9 +177,39 @@ public static void deleteComment(Long id, Long userId) {
...
+      photos = Photo.findAll();
+    }
+    else {
+      User user = User.findById(id);
+      photos = Photo.find("byOwner", user).fetch();
+    }
+    render(photos);
+  }
+  
+  public static void getPhoto(Long id) {
```
Comment (amshali) [inline app/controllers/Application.java]: Change the name of parameter

File: app/controllers/Photos.java [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+      renderText("");
+    }
+    else {
+      response.setContentTypeIfNotSet(photo.image.type());
+      renderBinary(photo.image.get());
+    }
+  }
+
+  public static void addPhoto(Photo photo) {
+    if (photo.image == null);{
```
Comment (ProCynic) [inline app/controllers/Photos.java]: BUG!  ; after if statement on this line.

Comment (amshali) [inline app/controllers/Photos.java]: Could you please send a revision?

-- Amin

On Mon, Feb 27, 2012 at 5:12 PM, Geoffrey Parker <
reply@reply.github.com

Comment (ProCynic) [inline app/controllers/Photos.java]: there you go.
-Geoffrey Parker

On Mon, Feb 27, 2012 at 5:17 PM, amshali <
reply@reply.github.com

File: app/models/Photo.java [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+import play.data.validation.*;
+
+
+@Entity
+public class Photo extends Model {
+  @Required
+  public Blob image;
+
+  @Required
+  public Date postedAt;
```
Comment (ProCynic) [inline app/models/Photo.java:16]: dateAdded

File: app/models/Photo.java [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+
+  public String caption;
```
Comment (ProCynic) [inline app/models/Photo.java:18]: caption

**Pull request merged**